cmd/dep/gom_importer.go

```diff
@@ -0,0 +1,259 @@
+			return err
+		}
+	}
+	defer f.Close()
+	br := bufio.NewReader(f)
+
+	g.goms = make([]gomPackage, 0)
+
+	n := 0
+	skip := 0
+	valid := true
+	var envs []string
+	for {
+		n++
+		lb, _, err := br.ReadLine()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+		line := strings.TrimSpace(string(lb))
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		name := ""
+		options := make(map[string]interface{})
+		var items []string
+		if gomReGroup.MatchString(line) {
+			envs = strings.Split(gomReGroup.FindStringSubmatch(line)[1], ",")
+			for i := range envs {
+				envs[i] = strings.TrimSpace(envs[i])[1:]
+			}
+			valid = true
+			continue
+		} else if gomReEnd.MatchString(line) {
+			if !valid {
+				skip--
+				if skip < 0 {
+					return fmt.Errorf("Syntax Error at line %d", n)
+				}
+			}
+			valid = false
+			envs = nil
+			continue
+		} else if skip > 0 {
+			continue
+		} else if gomReGom.MatchString(line) {
+			items = gomReGom.FindStringSubmatch(line)[1:]
+			name = unquote(items[0])
+			g.parseOptions(items[1], options)
+		} else {
+			return fmt.Errorf("Syntax Error at line %d", n)
+		}
+		if envs != nil {
+			options["group"] = envs
+		}
+		g.goms = append(g.goms, gomPackage{name, options})
+	}
+}
+
+// convert the gomfile into dep configuration files.
+func (g *gomImporter) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	g.logger.Println("Converting from Gomfile ...")
+
+	manifest := &dep.Manifest{
+		Constraints: make(gps.ProjectConstraints),
+	}
+	lock := &dep.Lock{}
+
+	for _, pkg := range g.goms {
+		// Obtain ProjectRoot. Required for avoiding sub-package imports.
+		ip, err := g.sm.DeduceProjectRoot(pkg.name)
+		if err != nil {
+			return nil, nil, err
+		}
+		pkg.name = string(ip)
+
+		// Check if it already existing in locked projects
+		if projectExistsInLock(lock, pkg.name) {
+			continue
+		}
+
+		rev := ""
+
+		if pkg.hasOption("branch") {
+			rev, _ = pkg.options["branch"].(string)
+		}
+		if pkg.hasOption("tag") {
+			rev, _ = pkg.options["tag"].(string)
+		}
+		if pkg.hasOption("commit") {
+			rev, _ = pkg.options["commit"].(string)
+		}
+
+		var pc gps.ProjectConstraint
+		pc.Ident = gps.ProjectIdentifier{ProjectRoot: gps.ProjectRoot(pkg.name)}
+
+		if rev != "" {
+			pi := gps.ProjectIdentifier{
+				ProjectRoot: gps.ProjectRoot(pkg.name),
+			}
+			revision := gps.Revision(rev)
+			version, err := lookupVersionForRevision(revision, pi, g.sm)
+			if err != nil {
+				warn := errors.Wrapf(err, "Unable to lookup the version represented by %s in %s. Falling back to locking the revision only.", rev, pi.ProjectRoot)
+				g.logger.Printf(warn.Error())
+				version = revision
+			}
+			feedback(version, pi.ProjectRoot, fb.DepTypeImported, g.logger)
+			lp := gps.NewLockedProject(pi, version, nil)
+			lock.P = append(lock.P, lp)
+		}
+		manifest.Constraints[pc.Ident.ProjectRoot] = gps.ProjectProperties{Source: pc.Ident.Source, Constraint: nil}
```

Comment:
Did you intend to always leave the constraint in the manifest `nil`? When the gomfile specifies `:tag => '1.0.0'`, or `:branch => 'master'` for example, both are valid constraints that can be figured out using `deduceConstraint`.